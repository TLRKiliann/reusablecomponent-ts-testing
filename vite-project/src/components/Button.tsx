import { Component } from 'react'

interface DataProps {
  isClicked: boolean
  description: string
}

export default class Button extends Component<DataProps> {
  state = {
      isClicked: false,
      description: "it's false"
  }

  handleClick = (event: MouseEvent<HTMButtonElement>) => {
    {this.state.isClicked 
      ? this.setState({isClicked: false, description: "it's false"})
      : this.setState({isClicked: true, description: "it's true"})}
  }

  render() {
    return(
      <>
      <button data-testid="btnid" onClick={this.handleClick} >
        my btn
      </button>
      <input
        data-testid="inputid"
        value={this.state.description} 
        onChange={this.handleClick}
        placeholder="inputplace"
      />
    </>
    )
  }
}
