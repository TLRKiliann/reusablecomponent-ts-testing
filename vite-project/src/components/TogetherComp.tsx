import { Component } from 'react'
import Header from './Header'
import Button from './Button'

type PrivateProps = {
  Header: React.ComponentType
  Button: React.ComponentType
}

export default class TogetherComp extends Component<PrivateProps> {
  render() {
    return(
      <>
        <Header />
        <Button />
      </>
    )
  }
}