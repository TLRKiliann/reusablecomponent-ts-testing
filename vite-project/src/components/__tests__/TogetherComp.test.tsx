import {beforeEach, describe, expect, test, it, vi} from 'vitest';
//import {fireEvent, render, screen, waitFor} from '@testing-library/react';
//import { act } from 'react-dom/test-utils';
import {create} from 'react-test-renderer';
import TogetherComp from '../TogetherComp.tsx'
import Header from '../TogetherComp.tsx'
import Button from '../TogetherComp.tsx'

describe('Seventh snapshot test', () => {
	test('testing TogetherComp component', () => {
		let tree = create(<TogetherComp />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})

describe('Eight snapshot test', () => {
	test('testing Header component', () => {
		let tree = create(<Header />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})

describe('Nine snapshot test', () => {
	test('testing Button component', () => {
		let tree = create(<Button />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})