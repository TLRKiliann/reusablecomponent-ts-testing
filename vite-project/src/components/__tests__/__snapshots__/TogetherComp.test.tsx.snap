// Vitest Snapshot v1

exports[`Eight snapshot test > testing Header component 1`] = `
[
  <div>
    Hello from Header
  </div>,
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>,
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />,
]
`;

exports[`Nine snapshot test > testing Button component 1`] = `
[
  <div>
    Hello from Header
  </div>,
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>,
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />,
]
`;

exports[`Seventh snapshot test > testing TogetherComp component 1`] = `
[
  <div>
    Hello from Header
  </div>,
  <button
    data-testid="btnid"
    onClick={[Function]}
  >
    my btn
  </button>,
  <input
    data-testid="inputid"
    onChange={[Function]}
    placeholder="inputplace"
    value="it's false"
  />,
]
`;
